### PR TITLE
feat(ckerc20): NNS proposal to upgrade ckERC20 ledger suites to support ICRC-21

### PR DESCRIPTION
**Pull request by gregorydemay**
Add NNS proposal to upgrade all ckERC20 ledger suites managed by the orchestrator to add support for ICRC-21

### CODE REVIEW

File: rs/ethereum/cketh/mainnet/orchestrator_upgrade_2024_08_05.md [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+
+Upgrade all ledger suites managed by the orchestrator to the latest version to add support for the [ICRC-21: Canister Call Consent Messages](https://github.com/dfinity/wg-identity-authentication/blob/fd846030109710cab67d9381485a73db424f2b07/topics/ICRC-21/icrc_21_consent_msg.md) standard.
+
+## Upgrade args
+
+```
+git fetch
+git checkout 3d0b3f10417fc6708e8b5d844a0bac5e86f3e17d
+cd rs/ethereum/ledger-suite-orchestrator
+didc encode -d ledger_suite_orchestrator.did -t '(OrchestratorArg)' '(variant { UpgradeArg = record {git_commit_hash = opt "3d0b3f10417fc6708e8b5d844a0bac5e86f3e17d"; ledger_compressed_wasm_hash = opt "8457289d3b3179aa83977ea21bfa2fc85e402e1f64101ecb56a4b963ed33a1e6"; index_compressed_wasm_hash = opt "eb3096906bf9a43996d2ca9ca9bfec333a402612f132876c8ed1b01b9844112a"; archive_compressed_wasm_hash = opt "5bd1f69540bd48493018e13bb5ad25aba75d59403ced1d5958bf718147228d31"}})'
```
Comment (gregorydemay) [inline rs/ethereum/cketh/mainnet/orchestrator_upgrade_2024_08_05.md:23]: `4449444c0e6b03ba9edbce040186dac7c60906c3cab9d20a0b6c05b59798d00202e898cad20405e9c2d6a70905a1eedde50a0598c6c2cf0b056e036c04e7aba21404cdd1f4e20404e4d8cce80b0494e2c9d00c046e7d6e716c03b59798d00207ffb08aab0809cb83d3c90f0a6e086c04e7aba2147dcdd1f4e2047de4d8cce80b7d94e2c9d00c7d6d686e686c0292bec7bd030cfdb5c9a9070d6c0299e7a4cf087db4e3ade809716c05c295a993017b9efeb9a40371f2c794ae037db1a38bf40d7191c9aafe0d7101000000014035626431663639353430626434383439333031386531336262356164323561626137356435393430336365643164353935386266373138313437323238643331012833643062336631303431376663363730386538623564383434613062616335653836663365313764014038343537323839643362333137396161383339373765613231626661326663383565343032653166363431303165636235366134623936336564333361316536014065623330393639303662663961343339393664326361396361396266656333333361343032363132663133323837366338656431623031623938343431313261`

File: rs/ethereum/cketh/mainnet/orchestrator_upgrade_2024_08_05.md [hunk 2]
```diff
@@ -0,0 +1,81 @@
+# Proposal to upgrade the ledger suite orchestrator canister
```
Comment (gregorydemay) [inline rs/ethereum/cketh/mainnet/orchestrator_upgrade_2024_08_05.md:1]:
```
~/bin/ic-admin \
    --use-hsm \
    --key-id 01 \
    --slot 0 \
    --pin $(cat ~/.hsm-pin) \
    --nns-url "https://ic0.app" \
    propose-to-change-nns-canister \
    --proposer 17212304975669116357 \
    --proposal-title "Add a new ckERC20 token: ckWSTETH" \
    --canister-id vxkom-oyaaa-aaaar-qafda-cai \
    --mode upgrade \
    --wasm-module-path ./ic-ledger-suite-orchestrator-canister.wasm.gz \
    --wasm-module-sha256 ccda8966249bc7d89f14968a242d6fb2c33d2363336303ec4f12977cab74c6f7 \
    --arg args.bin \
    --summary-file ./orchestrator_upgrade_2024_08_05.md
```

File: rs/ethereum/cketh/mainnet/orchestrator_upgrade_2024_08_05.md [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+
+## Wasm Verification
+
+Verify that the hash of the gzipped WASM matches the proposed hash.
+
+```
+git fetch
+git checkout 3d0b3f10417fc6708e8b5d844a0bac5e86f3e17d
+./gitlab-ci/container/build-ic.sh -c
+sha256sum ./artifacts/canisters/ic-ledger-suite-orchestrator-canister.wasm.gz
```
Comment (manudrijvers) [inline rs/ethereum/cketh/mainnet/orchestrator_upgrade_2024_08_05.md:69]: `ccda8966249bc7d89f14968a242d6fb2c33d2363336303ec4f12977cab74c6f7`

File: rs/ethereum/cketh/mainnet/orchestrator_upgrade_2024_08_05.md [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+```
+
+Verify that the hash of the gzipped WASM for the ledger, index and archive match the proposed hash.
+
+```
+git fetch
+git checkout 3d0b3f10417fc6708e8b5d844a0bac5e86f3e17d
+./gitlab-ci/container/build-ic.sh -c
+sha256sum ./artifacts/canisters/ic-icrc1-ledger-u256.wasm.gz
```
Comment (manudrijvers) [inline rs/ethereum/cketh/mainnet/orchestrator_upgrade_2024_08_05.md:78]: `8457289d3b3179aa83977ea21bfa2fc85e402e1f64101ecb56a4b963ed33a1e6`, matches args

File: rs/ethereum/cketh/mainnet/orchestrator_upgrade_2024_08_05.md [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+sha256sum ./artifacts/canisters/ic-icrc1-index-ng-u256.wasm.gz
```
Comment (manudrijvers) [inline rs/ethereum/cketh/mainnet/orchestrator_upgrade_2024_08_05.md:79]: `eb3096906bf9a43996d2ca9ca9bfec333a402612f132876c8ed1b01b9844112a`, matches args

File: rs/ethereum/cketh/mainnet/orchestrator_upgrade_2024_08_05.md [hunk 6]
```diff
@@ -0,0 +1,81 @@
...
+sha256sum ./artifacts/canisters/ic-icrc1-archive-u256.wasm.gz
```
Comment (manudrijvers) [inline rs/ethereum/cketh/mainnet/orchestrator_upgrade_2024_08_05.md:80]: `5bd1f69540bd48493018e13bb5ad25aba75d59403ced1d5958bf718147228d31`, matches args

**Review by manudrijvers — APPROVED** [review]

**Pull request merged**